source/blog/2016-09-21-running-tempest-on-rdo-openstack-newton.html.md

```diff
@@ -0,0 +1,124 @@
+
+Roll out a vm of [CentOS 7](https://www.centos.org/download/), Follow these steps:
+
+1. Install rdo-release-newton rpm
+
+        # yum -y install https://rdoproject.org/repos/openstack-newton/rdo-release-newton.rpm
+
+2. Update your centos vm and perform reboot.
+
+	    # yum -y update
+
+3. Install openstack-packstack
+
+	    # yum install -y openstack-packstack
+
+4. Run packstack by enabling RDO GA testing repo:
+
+	    # packstack --enable-rdo-testing=y --allinone
+
+   Once packstack installation is done, we are good to go ahead.
+
+**Install tempest and required tempest plugins**
+
+ 1. Install tempest
+
+	    # yum install openstack-tempest
+
+2. Install tempest plugins based on the openstack services installed and configured on deployment.
+
+   Packstack installs by default horizon, nova, neutron, keystone, cinder, swift, glance, ceilometer, aodh and gnocchi.
+   To find out what are the openstack components installed, just do a rpm query:
+
+	    # rpm -qa | grep openstack-*
+
+	OR you can use openstack-status command for the same.
+	Just grab the tempest plugins of these services and install it.
+
+	    # yum install python-glance-tests python-keystone-tests python-horizon-tests-tempest \
+	      python-neutron-tests python-cinder-tests python-nova-tests python-swift-tests \
+	      python-ceilometer-tests python-gnocchi-tests python-aodh-tests
+
+	OR you can simply download install_test_packages.py script to install tempest plugins and run it.
+	It will automatically installs the required tempest plugin test packages.
+
+	    # python /usr/share/openstack-tempest-*/tools/install_test_packages.py
+
+
+3. To find what are tempest plugins installed:
+
+	    # tempest list-plugins
+
+   Once done, you are ready to run tempest.
+
+**Configuring and Running tempest**
+
+1. source admin credentials and switch to normal user
+
+	    # source /root/keystonerc_admin
+
+	    # su <user>
+
+2. Create a directory from where you want to run tempest
+
+	    $ mkdir /home/$USER/tempest; cd /home/$USER/tempest
+
+3. Configure the tempest directory
+
+	    $ /usr/share/openstack-tempest-*/tools/configure-tempest-directory
+
+4.  Auto generate tempest configuration for your deployed openstack environment
+
+	    $ python tools/config_tempest.py --debug identity.uri $OS_AUTH_URL \
+          identity.admin_password  $OS_PASSWORD --create
+
+	It will automatically create all the required configuration in etc/tempest.conf
+
+5.  To list all the tests
+
+	    $ testr list-tests
+
+    OR
+
+	    $ ostestr -l
+
+6. To run tempest tests:
+
+	    $ ostestr
+
+7. For running api and scenario tests using ostestr and prints the slowest tests after test run
+
+	    $ ostestr --regex '(?!.*\[.*\bslow\b.*\])(^tempest\.(api|scenario))'
+
+8. To run specific tests:
+
+	    $ python -m testtools.run tempest.api.volume.v2.test_volumes_list.VolumesV2ListTestJSON
+
+	OR
+
+	    $ ostestr --pdb tempest.api.volume.v2.test_volumes_list.VolumesV2ListTestJSON
```

Comment:
How about mentioning that this would call python -m testtools.run under the hood.